src/mumble/GlobalShortcut_xdp.cpp

```diff
@@ -0,0 +1,189 @@
+// Copyright 2007-2022 Aleix Pol Gonzalez <aleixpol@kde.org>.
+// All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "GlobalShortcut_xdp.h"
+
+#include "Settings.h"
+#include "Global.h"
+#include "globalshortcuts_portal_interface.h"
+#include "portalsrequest_interface.h"
+#include "GlobalShortcutTypes.h"
+
+#include <QtGui/QWindow>
+
+Q_GLOBAL_STATIC_WITH_ARGS(OrgFreedesktopPortalGlobalShortcutsInterface, s_shortcutsInterface, (QLatin1String("org.freedesktop.portal.Desktop"),
+                                                                        QLatin1String("/org/freedesktop/portal/desktop"),
+                                                                        QDBusConnection::sessionBus()));
+
+bool GlobalShortcutXdp::isAvailable()
+{
+	return s_shortcutsInterface->isValid();
+}
+
+GlobalShortcutXdp::GlobalShortcutXdp() {
+	qDBusRegisterMetaType<XdpShortcuts>();
+	qDBusRegisterMetaType<QPair<QString, QVariantMap>>();
+
+	connect(s_shortcutsInterface, &OrgFreedesktopPortalGlobalShortcutsInterface::Activated, this, [this] (const QDBusObjectPath &session_handle, const QString &shortcut_id, qulonglong timestamp, const QVariantMap &options) {
+        Q_UNUSED(session_handle);
+        Q_UNUSED(timestamp);
+        Q_UNUSED(options);
+
+        foreach (GlobalShortcut *shortcut, qmShortcuts) {
+            if (shortcut_id == shortcut->objectName()) {
+                shortcut->triggered(true, shortcut_id);
+                shortcut->down(shortcut_id);
+            }
+        }
+    });
+    connect(s_shortcutsInterface, &OrgFreedesktopPortalGlobalShortcutsInterface::Deactivated, this, [this] (const QDBusObjectPath &session_handle, const QString &shortcut_id, qulonglong timestamp, const QVariantMap &options) {
+        Q_UNUSED(session_handle);
+        Q_UNUSED(timestamp);
+        Q_UNUSED(options);
+
+        foreach (GlobalShortcut *shortcut, qmShortcuts) {
+            if (shortcut_id == shortcut->objectName()) {
+                shortcut->triggered(false, shortcut_id);
+            }
+        }
+    });
+    connect(s_shortcutsInterface, &OrgFreedesktopPortalGlobalShortcutsInterface::ShortcutsChanged, this, [this] (const QDBusObjectPath &session_handle, const QList<QPair<QString,QVariantMap>> &shortcuts) {
+        Q_UNUSED(session_handle);
+        if (m_shortcuts != shortcuts) {
+            m_shortcuts = shortcuts;
+            Q_EMIT shortcutsChanged();
```

Comment:
```suggestion
            emit shortcutsChanged();
```
We typically just use the `emit` "keyword" directly